From fb0938d89f5b360104726f00fccf329ea2dbe64b Mon Sep 17 00:00:00 2001
From: Yankun Zheng <zyk@rock-chips.com>
Date: Thu, 7 Dec 2017 16:21:04 +0800
Subject: [PATCH] ARM: dts: rockchip: Modify pinctrl configuration for android
 things SOM.

fix the pinctrl default state is repeatedly defined
between rk3229-at-gva.dts and rk3229-at-som.dtsi

Change-Id: I117b3d97d446899ad7f35234df7c8dc0da60634e
Signed-off-by: Yankun Zheng <zyk@rock-chips.com>
---
 arch/arm/boot/dts/rk3229-at-gva.dts  |  98 ++---------------
 arch/arm/boot/dts/rk3229-at-som.dtsi | 150 +++++++++------------------
 2 files changed, 58 insertions(+), 190 deletions(-)

diff --git a/arch/arm/boot/dts/rk3229-at-gva.dts b/arch/arm/boot/dts/rk3229-at-gva.dts
index 3c4acf994b4d..12ca32f6e438 100644
--- a/arch/arm/boot/dts/rk3229-at-gva.dts
+++ b/arch/arm/boot/dts/rk3229-at-gva.dts
@@ -45,7 +45,7 @@
 #include "rk3229-at-som.dtsi"
 
 / {
-	model = "RK3229 ANDROID THINGS 3NOD Board";
+	model = "RK3229 ANDROID THINGS GVA Board";
 	compatible = "rockchip,rk3229-at-gva", "rockchip,rk3229";
 
 	sdio_pwrseq: sdio-pwrseq {
@@ -126,41 +126,13 @@
 };
 
 &pinctrl {
-	pinctrl-names = "default", "gpio_pwm0", "gpio_pwm1", "gpio_spdif", "gpio_irrx";
+	pinctrl-names = "default";
 	/*
-	 * sdmmc pins are defaultly set as gpio, if you want to enable
-	 * it, please remove below line.
+	 * sdmmc, pwm0, pwm2, ir, and spdif pins are defaultly set as gpio,
+	 * if you want to enable it, please remove below besides fixed_gpio.
 	 */
-	pinctrl-0 = <&gpio1b6_gpio &gpio1b7_gpio &gpio1c0_gpio
-			&gpio1c1_gpio &gpio1c2_gpio &gpio1c3_gpio
-			&gpio1c4_gpio &gpio1c5_gpio>;
-	/*
-	 * pwm0 pins are defaultly set as gpio, if you want to enable
-	 * it, please remove below line.
-	 */
-	pinctrl-1 = <&gpio3c5_gpio>;
-	/*
-	 * pwm1 pins are defaultly set as gpio, if you want to enable
-	 * it, please remove below line.
-	 */
-	pinctrl-2 = <&gpio1b4_gpio>;
-	/*
-	 * spdif pins are defaultly set as gpio, if you want to enable
-	 * it, please remove below line.
-	 */
-	pinctrl-3 = <&gpio3d7_gpio>;
-	/*
-	 * ir_rx pins are defaultly set as gpio, if you want to enable
-	 * it, please remove below line.
-	 */
-	pinctrl-4 = <&gpio1b3_gpio>;
-
-	ir-rx-gpio {
-		gpio1b3_gpio: gpio1b3-gpio {
-			rockchip,pins =
-					<1 11 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-	};
+	pinctrl-0 = <&sdmmc_gpio &pwm0_gpio &pwm2_gpio
+		     &ir_gpio &spdif_gpio &fixed_gpio>;
 
 	keys {
 		pwr_key: pwr-key {
@@ -168,67 +140,11 @@
 		};
 	};
 
-	pwm0-gpio {
-		gpio3c5_gpio: gpio3c5-gpio {
-			rockchip,pins =
-					<3 21 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-	};
-
-	pwm2-gpio {
-		gpio1b4_gpio: gpio1b4-gpio {
-			rockchip,pins =
-					<1 12 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-	};
-
 	sdio-pwrseq {
 		wifi_enable_h: wifi-enable-h {
 			rockchip,pins = <3 15 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
-
-	sdmmc-gpio {
-		gpio1b6_gpio: gpio1b6-gpio {
-			rockchip,pins =
-					<1 14 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-		gpio1b7_gpio: gpio1b7-gpio {
-			rockchip,pins =
-					<1 15 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		gpio1c0_gpio: gpio1c0-gpio {
-			rockchip,pins =
-					<1 16 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-		gpio1c1_gpio: gpio1c1-gpio {
-			rockchip,pins =
-					<1 17 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		gpio1c2_gpio: gpio1c2-gpio {
-			rockchip,pins =
-					<1 18 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		gpio1c3_gpio: gpio1c3-gpio {
-			rockchip,pins =
-					<1 19 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		gpio1c4_gpio: gpio1c4-gpio {
-			rockchip,pins =
-					<1 20 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		gpio1c5_gpio: gpio1c5-gpio {
-			rockchip,pins =
-					<1 21 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-	};
-
-	spdif-out-gpio {
-		gpio3d7_gpio: gpio3d7-gpio {
-			rockchip,pins =
-					<3 31 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-	};
 };
 
 &sdio {
@@ -241,7 +157,7 @@
 
 /*
  * If sdmmc is enabled on hardware, set status to okay.
- * and remove the according pinctrl nodes above.
+ * and remove the sdmmc_gpio from pinctrl-0 above.
  */
 &sdmmc {
 	bus-width = <4>;
diff --git a/arch/arm/boot/dts/rk3229-at-som.dtsi b/arch/arm/boot/dts/rk3229-at-som.dtsi
index 2de287d473d8..7ce43ec69969 100644
--- a/arch/arm/boot/dts/rk3229-at-som.dtsi
+++ b/arch/arm/boot/dts/rk3229-at-som.dtsi
@@ -137,113 +137,65 @@
 };
 
 &pinctrl {
-	pinctrl-names = "default";
-
-	pinctrl-0 = <&fixed0_gpio &fixed1_gpio &fixed2_gpio
-			&fixed3_gpio &fixed4_gpio &fixed5_gpio
-			&fixed6_gpio &fixed7_gpio &fixed8_gpio
-			&fixed9_gpio &fixed10_gpio &fixed11_gpio
-			&fixed12_gpio &fixed13_gpio &fixed14_gpio
-			&fixed15_gpio &fixed16_gpio &fixed17_gpio
-			&fixed18_gpio &fixed19_gpio &fixed20_gpio
-			&fixed21_gpio &fixed22_gpio>;
-
-	fixed_gpio {
-		fixed0_gpio: gpio2c3-gpio {
-			rockchip,pins =
-					<2 19 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed1_gpio: gpio2c2-gpio {
-			rockchip,pins =
-					<2 18 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed2_gpio: gpio2c1-gpio {
-			rockchip,pins =
-					<2 17 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed3_gpio: gpio2c0-gpio {
-			rockchip,pins =
-					<2 16 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed4_gpio: gpio2b7-gpio {
-			rockchip,pins =
-					<2 15 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed5_gpio: gpio2b6-gpio {
-			rockchip,pins =
-					<2 14 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed6_gpio: gpio2b5-gpio {
-			rockchip,pins =
-					<2 13 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed7_gpio: gpio2b4-gpio {
-			rockchip,pins =
-					<2 12 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed8_gpio: gpio2b3-gpio {
-			rockchip,pins =
-					<2 11 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed9_gpio: gpio2b2-gpio {
-			rockchip,pins =
-					<2 10 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed10_gpio: gpio2b1-gpio {
-			rockchip,pins =
-					<2 9 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed11_gpio: gpio2b0-gpio {
-			rockchip,pins =
-					<2 8 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed12_gpio: gpio1a7-gpio {
-			rockchip,pins =
-					<1 7 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed13_gpio: gpio1a3-gpio {
-			rockchip,pins =
-					<1 3 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed14_gpio: gpio1a0-gpio {
-			rockchip,pins =
-					<1 0 RK_FUNC_GPIO &pcfg_pull_down>;
+	sdmmc {
+		sdmmc_gpio: sdmmc-gpio {
+			rockchip,pins = <1 14 RK_FUNC_GPIO &pcfg_pull_none>,
+					<1 15 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 16 RK_FUNC_GPIO &pcfg_pull_none>,
+					<1 17 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 18 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 19 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 20 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 21 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
-		fixed15_gpio: gpio1b0-gpio {
-			rockchip,pins =
-					<1 8 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		fixed16_gpio: gpio0c1-gpio {
-			rockchip,pins =
-					<0 17 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		fixed17_gpio: gpio0d0-gpio {
-			rockchip,pins =
+	};
+
+	fixed_io {
+		fixed_gpio: fixed-gpio {
+			rockchip,pins = <2 19 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 18 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 17 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 16 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 15 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 14 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 13 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 12 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 11 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 10 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 9 RK_FUNC_GPIO &pcfg_pull_none>,
+					<2 8 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 7 RK_FUNC_GPIO &pcfg_pull_none>,
+					<1 3 RK_FUNC_GPIO &pcfg_pull_none>,
+					<1 0 RK_FUNC_GPIO &pcfg_pull_none>,
+					<1 8 RK_FUNC_GPIO &pcfg_pull_up>,
+					<0 17 RK_FUNC_GPIO &pcfg_pull_up>,
 					<0 24 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
-		fixed18_gpio: gpio3c4-gpio {
-			rockchip,pins =
-					<3 20 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-		fixed19_gpio: gpio2d1-gpio {
-			rockchip,pins =
-					<2 25 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		fixed20_gpio: gpio2d0-gpio {
-			rockchip,pins =
-					<2 24 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-		fixed21_gpio: gpio2c6-gpio {
-			rockchip,pins =
-					<2 22 RK_FUNC_GPIO &pcfg_pull_up>;
+	};
+
+	ir {
+		ir_gpio: ir-gpio {
+			rockchip,pins = <1 11 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
-		fixed22_gpio: gpio2c7-gpio {
-			rockchip,pins =
-					<2 23 RK_FUNC_GPIO &pcfg_pull_up>;
+	};
+
+	pwm0 {
+		pwm0_gpio: pwm0-gpio {
+			rockchip,pins = <3 21 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
+	};
 
+	pwm2 {
+		pwm2_gpio: pwm2-gpio {
+			rockchip,pins = <1 12 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
 	};
 
+	spdif {
+		spdif_gpio: spdif-gpio {
+			rockchip,pins = <3 31 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
 };
 
 &rockchip_suspend {
-- 
2.35.3

